src/Plugin/Block/EzContentUserShortcutBlock.php

```diff
@@ -72,46 +72,54 @@ public function build() {
 
     // Getting current user role.
     $currentUserRoles = $this->account->getRoles();
-    if (!in_array('administrator', $currentUserRoles)) {
-      $currentUserId = $this->account->id();
-      $currentUsername = $this->account->getUsername();
-      $currentShortcutset = $this->entityTypeManager
-        ->getStorage('shortcut_set')
-        ->loadByProperties(['id' => 'shortcut_set_' . $currentUsername . '_' . $currentUserId]);
-      // Getting current user shortcut set.
-      $shortcutSet = $this->entityTypeManager
-        ->getListBuilder('shortcut_set');
-      if (!$currentShortcutset) {
-        // Creating default shortcut set for new user.
-        $set = $shortcutSet->getStorage()->create([
-          'id' => 'shortcut_set_' . $currentUsername . '_' . $currentUserId,
-          'label' => $this->t('Default Shortcut Set for @currentUsername', ['@currentUsername' => $currentUsername]),
-        ]);
-        $set->save();
+    if ($this->account->hasPermission('ezcontent user based shortcut')) {
+      if (!in_array('administrator', $currentUserRoles)) {
+        $currentUserId = $this->account->id();
+        $currentUsername = $this->account->getUsername();
+        $currentShortcutset = $this->entityTypeManager
+          ->getStorage('shortcut_set')
+          ->loadByProperties(['id' => 'shortcut_set_' . $currentUsername . '_' . $currentUserId]);
+        // Getting current user shortcut set.
+        $shortcutSet = $this->entityTypeManager
+          ->getListBuilder('shortcut_set');
+        if (!$currentShortcutset) {
+          // Creating default shortcut set for new user.
+          $set = $shortcutSet->getStorage()->create([
+            'id' => 'shortcut_set_' . $currentUsername . '_' . $currentUserId,
+            'label' => $this->t('Default Shortcut Set for @currentUsername', ['@currentUsername' => $currentUsername]),
+          ]);
+          $set->save();
+        }
+        $shortcutSetId = 'shortcut_set_' . $currentUsername . '_' . $currentUserId;
+        $shortcutObj = $this->getUserShortcutLinks($shortcutSetId);
       }
-      $shortcutSetId = 'shortcut_set_' . $currentUsername . '_' . $currentUserId;
-      $shortcutObj = $this->getUserShortcutLinks($shortcutSetId);
+      else {
+        $shortcutSetId = 'default';
+        $shortcutObj = $this->getUserShortcutLinks($shortcutSetId);
+      }
+
+      return [
+        '#theme' => 'user_shortcut_block_template',
+        '#data' => $shortcutObj,
+        '#cache' => [
+          'tags' => ['config:shortcut.set.' . $shortcutSetId],
+        ],
+      ];
     }
     else {
-      $shortcutSetId = 'default';
-      $shortcutObj = $this->getUserShortcutLinks($shortcutSetId);
+      return [
```

Comment:
we can handle empty behaviour as below that would not render block if empty.`/**
   * {@inheritdoc}
   */
  public function build() {
    $build = [];
    // Getting current user role.
    $currentUserRoles = $this->account->getRoles();
    if ($this->account->hasPermission('ezcontent user based shortcut')) {
      if (!in_array('administrator', $currentUserRoles)) {
        $currentUserId = $this->account->id();
        $currentUsername = $this->account->getUsername();
        $currentShortcutset = $this->entityTypeManager
          ->getStorage('shortcut_set')
          ->loadByProperties(['id' => 'shortcut_set_' . $currentUsername . '_' . $currentUserId]);
        // Getting current user shortcut set.
        $shortcutSet = $this->entityTypeManager
          ->getListBuilder('shortcut_set');
        if (!$currentShortcutset) {
          // Creating default shortcut set for new user.
          $set = $shortcutSet->getStorage()->create([
            'id' => 'shortcut_set_' . $currentUsername . '_' . $currentUserId,
            'label' => $this->t('Default Shortcut Set for @currentUsername', ['@currentUsername' => $currentUsername]),
          ]);
          $set->save();
        }
        $shortcutSetId = 'shortcut_set_' . $currentUsername . '_' . $currentUserId;
        $shortcutObj = $this->getUserShortcutLinks($shortcutSetId);
      }
      else {
        $shortcutSetId = 'default';
        $shortcutObj = $this->getUserShortcutLinks($shortcutSetId);
      }

       $build =  [
        '#theme' => 'user_shortcut_block_template',
        '#data' => $shortcutObj,
        '#cache' => [
          'tags' => ['config:shortcut.set.' . $shortcutSetId],
        ],
      ];
      return $build;
    }
    return $build;
  }`